### ISSUE TITLE
New Component: Switch Chain Modal

### ISSUE THREAD

**Issue by 0xTijan** [web3ui/web3uikit#865]
Many dApps require a functionality for user to be able to switch chains. Even though this component wouldn't use Moralis it could still be a nice useful component for the web3uikit.

**Comment by BillyG83** [comment]
@0xTijan can you outline how this would work?

**Comment by 0xTijan** [comment]
Hey Bill 👋

This would be a dropdown or modal (not sure yet) like in the 1inch or gnosis safe apps for example. When user selects a chain they want to switch to, Metamask pops up for user to approve that. The component would need to have at least 2 props:
- `currentChain`
- and `chainList` - array of wanted chains (EVM) to be in the dropdown or modal

But after looking at a dropdown component this new switch chain component maybe isn't needed at all? 😅

1inch example
![image](https://user-images.githubusercontent.com/88556617/198693040-d2d7a642-8e0f-40a2-974e-dbb608af9a2c.png)


Gnosis Safe example
![image](https://user-images.githubusercontent.com/88556617/198693081-60848d18-cb23-424a-8f9d-1401680ff1d1.png)

**Comment by AbhinavMV** [comment]
It would have been a nice addition but since moralis servers are going to be deprecated soon i don't think adding this to web3 package is worth it sorry 😅
but we can add it to the new package once we have it ready in future what do you think?

**Comment by 0xTijan** [comment]
Yeah, makes sense 😃

**Comment by BillyG83** [comment]
yeah i'm pretty close to adding the new web3 section @0xTijan , just some final things to workout about how to use NextJS pages with server side rendering in Storybook and make it useful to export to the end dev. Then this will be very nice. I do not think we need more UI for it, maybe just an array of chains would be passed and we would render the @AbhinavMV Select Component with the correct text and logos, then on change it could return the chain ID

**Comment by BillyG83** [comment]
i marked this blocked for now until i have the new slice ready, but you could actually add it to the `Form` section for now if you want @0xTijan

**Comment by BillyG83** [comment]
FYI @0xTijan i have added this to web3

I am using this tag to collect idea for the next iteration of the web3 section of the kit and i think this is a great idea! 🤩
